ntbs-service-unit-tests/Services/NtbsSearchBuilderTest.cs

```diff
@@ -1,467 +1,307 @@
+            var result =
+                ((INtbsSearchBuilder)_builder.FilterByPartialNotificationDate(
+                    new PartialDate() {Day = "1", Month = "1", Year = "3000"})).GetResult().ToList();
 
-                Assert.Empty(result);
-            }
+            Assert.Empty(result);
         }
 
         [Fact]
-        public async Task SearchByPartialDob_ReturnsMatchOnDob()
+        public void SearchByPartialDob_ReturnsMatchOnDob()
         {
-            using (var context = new NtbsContext(ContextOptions))
-            {
-                await context.Database.EnsureDeletedAsync();
-                await context.Database.EnsureCreatedAsync();
-                var builder = SetupTestDatabase(context);
-                var result =
-                    ((INtbsSearchBuilder)builder.FilterByPartialDob(new PartialDate()
-                    {
-                        Day = "1", Month = "1", Year = "1990"
-                    })).GetResult().ToList();
+            var result =
+                ((INtbsSearchBuilder)_builder.FilterByPartialDob(new PartialDate()
+                {
+                    Day = "1", Month = "1", Year = "1990"
+                })).GetResult().ToList();
 
-                Assert.Single(result);
-                Assert.Equal(new DateTime(1990, 1, 1), result.FirstOrDefault().PatientDetails.Dob);
-            }
+            Assert.Single(result);
+            Assert.Equal(new DateTime(1990, 1, 1), result.FirstOrDefault().PatientDetails.Dob);
         }
 
         [Fact]
-        public async Task SearchByPartialDob_NoResults()
+        public void SearchByPartialDob_NoResults()
         {
-            using (var context = new NtbsContext(ContextOptions))
-            {
-                await context.Database.EnsureDeletedAsync();
-                await context.Database.EnsureCreatedAsync();
-                var builder = SetupTestDatabase(context);
-                var result =
-                    ((INtbsSearchBuilder)builder.FilterByPartialDob(new PartialDate()
-                    {
-                        Day = "1", Month = "1", Year = "3000"
-                    })).GetResult().ToList();
+            var result =
+                ((INtbsSearchBuilder)_builder.FilterByPartialDob(new PartialDate()
+                {
+                    Day = "1", Month = "1", Year = "3000"
+                })).GetResult().ToList();
+
+            Assert.Empty(result);
+        }
+    }
+
+    public class DatabaseFixture : IDisposable
+    {
+        public DatabaseFixture()
+        {
+            Context = SetupTestDatabase(
+                new NtbsContext(new DbContextOptionsBuilder<NtbsContext>()
+                    .UseInMemoryDatabase(nameof(NtbsSearchBuilderTest))
+                    .Options
+                )
+            );
+        }
 
-                Assert.Empty(result);
-            }
+        public void Dispose()
+        {
+            Context.Dispose();
         }

```

Comment:
This much of the class looks like it could well be useful elsewhere. So perhaps we could pull it out into a Helper, and call SetupTestDatabase in this class from the constructor instead?

Reply:
The problem with calling SetupTestDatabase in the constructor is that xUnit.net creates a new instance of the test class for every test that is run, so any code which is placed into the constructor of the test class will be run for every single test. This means that we'd have to do cleanup for every test, which is what we are trying to avoid. The code that only runs once is the code in DatabaseFixture

Reply:
Ah, that's annoying! In that case I guess the "right" thing to do would be to make `DatabaseFixture` an abstract base class, and have different sub-classes for each test-suite that use it. But there's definitely an argument that the best time to do that refactor would be at the point that we need it. So I'm happy for you to merge.